includes/class-sensei-modules.php

```diff
@@ -805,8 +808,110 @@ public function module_archive_description() {
 				echo '<p class="status ' . esc_attr( $class ) . '">' . esc_html( $status ) . '</p>';
 			}
 
-			echo '<p class="archive-description module-description">' . wp_kses_post( apply_filters( 'sensei_module_archive_description', nl2br( $module->description ), $module->term_id ) ) . '</p>';
+			if ( $this->can_view_module_content( $module, $course_id, $user_id ) ) {
+				echo '<p class="archive-description module-description">' . wp_kses_post( apply_filters( 'sensei_module_archive_description', nl2br( $module->description ), $module->term_id ) ) . '</p>';
+			}
+		}
+	}
+
+	/**
+	 * Check if we can view module content.
+	 *
+	 * @param WP_Term $module    Module term object. Defaults to the currently queried term.
+	 * @param int     $course_id Course post ID. May not be set if not viewing module in course context.
+	 * @param int     $user_id   User ID. Defaults to currently logged in user ID.
+	 *
+	 * @return bool
+	 */
+	public function can_view_module_content( WP_Term $module = null, $course_id = null, $user_id = null ) {
+		$can_view_module_content = false;
+
+		if ( null === $module ) {
+			$module = get_queried_object();
+		}
+
+		if ( ! $module instanceof WP_Term ) {
+			return false;
+		}
+
+		if ( null === $user_id ) {
+			$user_id = get_current_user_id();
+		}
+
+		if ( ! sensei_is_login_required() ) {
+			$can_view_module_content = true;
+		} elseif ( $course_id && Sensei()->course->can_access_course_content( $course_id, $user_id, 'module' ) ) {
+			$can_view_module_content = true;
+		}
+
+		/**
+		 * Filter if the user can view module content.
+		 *
+		 * @since 3.0.0
+		 *
+		 * @param bool $can_view_module_content True if they can view module content.
+		 * @param int  $module_term_id          Module term ID.
+		 * @param int  $course_id               Course post ID.
+		 * @param int  $user_id                 User ID.
+		 */
+		return apply_filters( 'sensei_can_user_view_module', $can_view_module_content, $module->term_id, $course_id, $user_id );
+	}
+
+	/**
+	 * Outputs the module course sign-up link.
+	 *
+	 * @access private
+	 * @since 3.0.0
+	 */
+	public function course_signup_link() {
```

Comment:
Seems at a glance like we could potentially extract common functionality between this and `Sensei_Lesson::course_signup_link`. But there might be just enough small differences to make it complicated. Do you think it would be worth it?

Reply:
It might over complicate it, but I could try. I considered doing this but then shied from it when I thought all the strings and hooks would turn dynamic.

Reply:
I started working on this, but it seemed too complicated for now. It would just be the message generation, really, and I think the internationalization would make it too complicated. We couldn't do "before you start the %s" and then translate quiz, lesson, or module above because `the` would probably be different for certain languages.

Reply:
Ok yeah I thought something like that might overcomplicate it. Let's not worry about it then 🙂